proteobench/io/params/alphadia.py

```diff
@@ -89,22 +110,53 @@ def parse_section(
         except:
             break
 
+        if not isinstance(line_dict_next, dict):
+            continue
+
         # If the next line is start of new section again
         if level_next > level_prev:
-            # Get the subsection
-
-            subsection, _, next_line = parse_section(
-                line=parse_line(next_line),
-                line_generator=line_generator,
-            )
-            # Add this subsection to new section
-            # A new line is already outputted so continue
-            section[header_prev] = subsection
-            continue
+            if header_prev in ["precursor_len", "precursor_charge", "precursor_mz", "fragment_mz"]:
+                if current_header is None or current_header != header_prev:
+                    nested_values = []
+                    current_header = header_prev
+
+                # Collect all values under this nested section
+                value = list(line_dict_next.keys())[0].split()[0]  # Extract value before space
+                nested_values.append(int(value))
+
+                # If "user defined", overwrite the default
+                if "(user defined)" in list(line_dict_next.keys())[0]:
+                    nested_values.pop(-2)
+
+                # Save the values in the section
+                section[header_prev] = nested_values
+
+                try:
+                    next_line = next(line_generator)
+                    continue
+                except:
```

Comment:
Consider putting this if in a different function. Add specific error to the except. IOError?